client.go

```diff
@@ -16,4 +22,5 @@ type Client interface {
 
 	SetRetryCount(count int)
 	SetRetrier(retrier Retriable)
+	SetCustomHTTPClient(customHTTPClient Doer)
```

Comment:
Is there any specific reason to accept any `Doer` instead of just `*http.Client` ?

Reply:
@rShetty this is so that one can use a custom `HTTPClient` here, not just the one defined in one's code, but one that is defined in any imported library etc. 

This [blog post](https://www.0value.com/let-the-doer-do-it) talks about the advantages of accepting `Doer` over `*http.Client` (tl;dr composability). Also, I have seen packages use `Doer` to accept custom clients like [fetchbot](https://github.com/PuerkitoBio/fetchbot/blob/master/fetch.go#L54)

Reply:
👍